internal/mq/msgstream/mqwrapper/kafka/kafka_consumer.go

```diff
@@ -116,7 +117,9 @@ func (kc *Consumer) Chan() <-chan mqwrapper.Message {
 		panic("failed to chan a kafka consumer without assign")
 	}
 	kc.chanOnce.Do(func() {
+		kc.wg.Add(1)
 		go func() {
+			defer kc.wg.Done()
 			for {
 				select {
 				case <-kc.closeCh:
```

Comment:
it doesn't need to wait for the closing goroutine to finish, because the close is an async operation for the consumer.

Reply:
will it be a problem that the legacy goroutine still calling `kc.c.ReadMessage` while internal client is closed?

Reply:
yes, but it could happen that close client happens first, and ReadMessage still in progress. This waitgroup is meant to prevent this situation